moon/controller_excavator_round2.py

```diff
@@ -97,81 +129,215 @@ def vslam_reset_time(response):
             self.publish("bucket_drop", [math.pi, 'reset'])
 
             while self.mount_angle is None or abs(normalizeAnglePIPI(math.pi - self.mount_angle)) > 0.2:
-                self.wait(timedelta(seconds=1))
-
+                try:
+                    self.wait(timedelta(seconds=1))
+                except:
+                    pass
 
             self.send_request('vslam_reset', vslam_reset_time)
 
             while vol_list is None or not self.true_pose:
-                self.wait(timedelta(seconds=1))
+                try:
+                    self.wait(timedelta(seconds=1))
+                except:
+                    pass
 
             # calculate position behind the excavator to send hauler to, hauler will continue visually from there
             v = np.asmatrix(np.asarray([self.xyz[0], self.xyz[1], 1]))
             c, s = np.cos(self.yaw), np.sin(self.yaw)
             R = np.asmatrix(np.array(((c, -s, 0), (s, c, 0), (0, 0, 1))))
             T = np.asmatrix(np.array(((1, 0, -8),(0, 1, 0),(0, 0, 1))))
             pos =  np.dot(R, np.dot(T, np.dot(R.I, v.T)))
-            self.send_request('external_command hauler_1 goto %.1f %.1f %.2f' % (pos[0], pos[1], self.yaw))
+            #self.send_request('external_command hauler_1 goto %.1f %.1f %.2f' % (pos[0], pos[1], self.yaw))
```

Comment:
is it OK, that these requests are not commented out?

Reply:
I was testing three different strategies to get the robots into starting position:
1) tell hauler to move to a certain spot (goto)
2) tell hauler to align itself behind the excavator (which implies hauler will find the excavator visually)
3) tell hauler to point itself in certain direction

We are using (3) now but I may still go back to one of the other ones based on more testing. The problem with (2) was that many times hauler would find processing plant instead of excavator. However, I have tighten the artifact detector and maybe it's not a problem anymore, this method saves true_pose call for hauler that can be used later for re-localization